core/modules/catalog/helpers/index.ts

```diff
@@ -567,11 +567,11 @@ export function attributeImages (product) {
 
 export function configurableChildrenImages (product) {
   let configurableChildrenImages = []
-  if (product.configurable_children && product.configurable_children.length > 0) {
+  if (product.configurable_children && product.configurable_children.length > 0 && this.childHasImage(product.configurable_children)) {
```

Comment:
if childHasImage method would have default value then here it could only look like this
```suggestion
  if  (childHasImage(product.configurable_children)) {
```